src/instance.ts

```diff
@@ -16,13 +16,110 @@
 
 import arrify = require('arrify');
 const common = require('@google-cloud/common-grpc');
-import {ServiceObjectConfig} from '@google-cloud/common';
+import {ServiceObjectConfig, GetConfig, Metadata} from '@google-cloud/common';
 import {paginator} from '@google-cloud/paginator';
 import {promisifyAll} from '@google-cloud/promisify';
 import * as extend from 'extend';
-import * as is from 'is';
 import snakeCase = require('lodash.snakecase');
 import {Database} from './database';
+import {google as dbInstanceAdminClient} from '../proto/spanner_instance_admin';
+import {google as dbDatabaseAdminClient} from '../proto/spanner_database_admin';
+import {Operation as GaxOperation} from 'google-gax/build/src/longRunningCalls/longrunning';
+import {SessionPoolOptions, SessionPoolInterface} from './session-pool';
+import {SessionPool, Spanner} from '.';
+import {ServiceError} from 'grpc';
+
+export type IDatabase = dbDatabaseAdminClient.spanner.admin.database.v1.IDatabase;
+export type IInstance = dbInstanceAdminClient.spanner.admin.instance.v1.IInstance;
+export type LongrunningIOperation = dbInstanceAdminClient.longrunning.IOperation;
+
+export type CreateInstanceResponse = [
+  Instance,
+  GaxOperation,
+  LongrunningIOperation
+];
+export type CreateDatabaseResponse = [
+  Database,
+  GaxOperation,
+  LongrunningIOperation
+];
+export type DeleteInstanceResponse = [dbInstanceAdminClient.protobuf.IEmpty];
+export type ExistsInstanceResponse = [boolean];
+export type GetInstanceResponse = [Instance, IInstance];
+export type GetDatabasesResponse = [Database[], LongrunningIOperation];
+export type LongrunningOperationResponse = [
+  GaxOperation,
+  LongrunningIOperation
+];
+
+export interface CreateDatabaseOptions {
```

Comment:
I think we want to extend `ICreateDatabaseRequest` here

Reply:
Done!!